htlcswitch/switch.go

```diff
@@ -460,6 +460,18 @@ func (s *Switch) UpdateForwardingPolicies(
 	s.indexMtx.RUnlock()
 }
 
+// IsForwardedHTLC checks for a given channel and htlc index if it is related
+// to an opened circuit that represents a forwarded payment.
+func (s *Switch) IsForwardedHTLC(chanID lnwire.ShortChannelID,
+	htlcIndex uint64) bool {
+
+	circuit := s.circuits.LookupOpenCircuit(channeldb.CircuitKey{
+		ChanID: chanID,
+		HtlcID: htlcIndex,
+	})
+	return circuit != nil && circuit.Incoming.ChanID != hop.Source
+}
```

Comment:
This function isn't covered by the test above, but is important for proper behavior of the mechanism introduced in this pr. A low-effort option could be to add a few asserts in the existing switch tests.

Reply:
I took the low effort path which seems to me also the right one.

Reply:
Yes, very compact :+1: